CMakeLists.txt

```diff
@@ -262,8 +262,8 @@ if(BUILD_DEV)
 else()
     set(MIOPEN_BUILD_DEV 0)
     set(MIOPEN_SYSTEM_DB_PATH "${CMAKE_INSTALL_PREFIX}/${DATA_INSTALL_DIR}/db" CACHE PATH "Default path of system db files")
-    set(MIOPEN_USER_DB_PATH "~/.config/miopen/" CACHE STRING "Default path of user db files")
-    set(MIOPEN_CACHE_DIR "~/.cache/miopen/" CACHE STRING "")
+    set(MIOPEN_USER_DB_PATH "~/.config/miopen.${MIOpen_VERSION_MAJOR}_${MIOpen_VERSION_MINOR}_${MIOpen_VERSION_PATCH}_${MIOpen_VERSION_TWEAK}/" CACHE STRING "Default path of user db files")
+    set(MIOPEN_CACHE_DIR "~/.cache/miopen.${MIOpen_VERSION_MAJOR}_${MIOpen_VERSION_MINOR}_${MIOpen_VERSION_PATCH}_${MIOpen_VERSION_TWEAK}/" CACHE STRING "")
```

Comment:
The version for the cache happens [here](https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/src/binary_cache.cpp#L63) not in cmake. That is where the tweak should be added.